carma_wm_ros2/src/WMListener.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <new>
+#include <carma_wm_ros2/WMListener.hpp>
+#include "WMListenerWorker.hpp"
+
+namespace carma_wm
+{
+  // @SONAR_STOP@
+WMListener::WMListener(const rclcpp::NodeOptions &options, bool multi_thread)
```

Comment:
Readme examples needs to be updated to show how the user can pass in the node options